### PR TITLE
DONE - Add Confidentiality Agreement

### PR DESCRIPTION
**Pull request by markottaviani**
Okay, I can't figure out what I'm doing wrong here...

### CODE REVIEW

**Comment by markottaviani** [comment]
Keep getting this error:
screenshot_2016-03-26-16-10-53 833

**Comment by markottaviani** [comment]
We already have an "agreement" with is a polymorphic "has_one" of an Attachment class.  Trying to add a second but call it a "confidentiality_agreement", also a "has_one" of an Attachment class.

**Comment by TheDudeWithTheThing** [comment]
Are you initializing that anywhere? On update it might be ok when one exists but on a brand new user it will have none and af.object is nil. :bomb:

**Comment by markottaviani** [comment]
Jackpot!

**Comment by markottaviani** [comment]
Actually.. maybe not hummm

**Comment by markottaviani** [comment]
Okay hitting this issue now, so I'll need to apply the fix mentioned here next.  

http://stackoverflow.com/questions/20966110/rails-4-multiple-polymorphic-has-one-relations-not-working

**Comment by TheDudeWithTheThing** [comment]
multiple `has_one`, one per type you want to support?  That sounds fine, I think you'll need a hidden var to pass the type up in the form too

**Comment by markottaviani** [comment]
Check this out, I think we're done!

File: app/views/users/_form.html.erb [hunk 1]
```diff
@@ -100,7 +100,22 @@
...
+          </div>
+        </div>
+      <% end %>
+
+      <% if is_admin? %>
+        <div class="control-group">
+          <%= f.label :confidentiality_agreement, 'Confidentiality Agreement', :class => 'control-label' %>
+          <div class="controls">
+          <%= f.fields_for :confidentiality_agreement do |af| %>
+            <% if af.object.new_record? %>
```
Comment (TheDudeWithTheThing) [inline app/views/users/_form.html.erb]: could clean this up a pinch

```
<% if af.object.persisted? %>
  <span>Confidentiality Agreement on File, overwrite?</span> <%= af.file_field :attachment %>
<% end %>
<%= af.file_field :attachment %>
```

File: db/migrate/20160412010043_set_agreement_types_for_attachable.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class SetAgreementTypesForAttachable < ActiveRecord::Migration
+  def up
+    execute "UPDATE attachments SET agreement_type = 'foster' WHERE attachable_type = 'User';"
```
Comment (TheDudeWithTheThing) [inline db/migrate/20160412010043_set_agreement_types_for_attachable.rb]: why the raw SQL?

```ruby
  Attachment.where(attachable_type: 'User').update_all(agreement_type: 'foster')
```

**Comment by TheDudeWithTheThing** [comment]
Good stuff, just some "the more you know" type comments.

:+1:

**Comment by markottaviani** [comment]
Added your suggestions!  Does the ```object.persisted``` stuff look alright?

File: app/views/users/_form.html.erb [hunk 3]
```diff
@@ -97,11 +97,11 @@
           <%= f.label :agreement, 'Foster Agreement', :class => 'control-label' %>
           <div class="controls">
           <%= f.fields_for :agreement do |af| %>
-            <% if af.object.new_record? %>
+            <% if af.object.persisted? %>
               <%= af.file_field :attachment %>
```
Comment (TheDudeWithTheThing) [inline app/views/users/_form.html.erb]: you'll have 2 input fields if you leave this no?  you basically want the conditional to show the overwrite? if one exists already right?  trying to remove having to have the line `<%= af.file_field :attachment %>` twice

File: db/migrate/20160412010043_set_agreement_types_for_attachable.rb [hunk 4]
```diff
@@ -1,9 +1,9 @@
 class SetAgreementTypesForAttachable < ActiveRecord::Migration
   def up
-    execute "UPDATE attachments SET agreement_type = 'foster' WHERE attachable_type = 'User';"
+    Attachment.where(attachable_type: 'User').update_all(agreement_type: 'foster')
   end
 
   def down
-    raise ActiveRecord::IrreversibleMigration
+    Attachment.where(attachable_type: 'User').update_all(agreement_type: nil)
```
Comment (TheDudeWithTheThing) [inline db/migrate/20160412010043_set_agreement_types_for_attachable.rb]: nice `down` !

File: app/views/users/_form.html.erb (hunk 3 above)
Comment (markottaviani) [inline app/views/users/_form.html.erb]: whoops, yeah.  updated!

**Comment by markottaviani** [comment]
Okay, I think we're golden

**Pull request merged**